proposals/NNNN-unlock-existentials-with-fixed-associated-types.md

```diff
@@ -0,0 +1,237 @@
+
+Now, a protocol is considered unusable as a Type when it:
+
+1. Includes at least one associated type or `Self` requirement - which may have been inherited - and
+2. if at least one inherited associated type requirement - if there is any - is _not_ specified.
+
+
+### Protocols that would be Usable as Types
+
+1. Fixed Associated Types
+```swift
+protocol AB { 
+    associatedtype A
+    associatedtype B
+}
+
+protocol FixedAB: AB
+    where A == String, B == String {}
+    
+let foo: FixedAB 
+// ✅ 
+```
+> **_NOTE:_** Note that both `A` and `B` were fixed in order for `FixedAB` to be usable as a Type.
+
+2. Fixed Associated Types in Composition
+```swift
+protocol C { 
+    associatedtype C
+}
+
+protocol FixedC: C
+    where C == String {}
+
+typealias FixedABC = FixedAB & FixedC
+
+let foo: FixedABC 
+// ✅ 
+```
+
+3. Fixed Associated Types in Composition with a Class
+```swift
+class Class {}
+
+typealias FixedABAndClass = FixedAB & Class
+
+let foo: FixedABAndClass 
+// ✅ 
+```
+
+### Protocols that would NOT be Usable as Types
+
+1. Fixed Associated Types with `Self` Requirement
+```swift
+protocol ABAndSelf {
+    associatedtype A
+    associatedtype B
+    
+    func foo(a: Self)
+}
+
+protocol FixedABAndSelf: ABAndSelf
+    where A == String, B == String {} 
+    
+let foo: FixedABAndSelf 
+// ❌ Explanation: `Self` cannot 
+// be specified; therefore it
+// cannot be used as a Type.
+```
+
+
+### Syntax 
+
+There’s no syntax change. This change is rather semantic, easing existing restrictions regarding the use of protocols as Types.
+
+
+## Source compatibility
+
+This is an additive change with no impact on source compatibility.
+
+
+## Effect on ABI stability
+
+TBA
+
+## Effect on API resilience
+
+With this proposal, Library Authors should be more considerate when adding more associated types to their publicly exposed protocols. That's because protocols that inherit associated type or `Self` requirements would - under this proposal - gain the ability to be used as Types - [under the right circumstances](#which-protocols-would-not-be-usable-as-types)).
+
+For instance, if we added another associated type to `Identifiable`, a _hypothetical_ `User` protocol in the Standard Library would become unusable as a Type, causing source breakage for clients and potentially inside the module itself. Moreover, protocols inheriting `Identifiable` outside of the Standard Library would also be unable to be used as Types aggravating the problem as a result.
+
+To reflect these new guidelines the 7th rule for 'allowed' changes ([protocol section](https://github.com/apple/swift/blob/master/docs/LibraryEvolution.rst#protocols)) will be removed. The rule to be changed, states that: 
+> A new associatedtype requirement may be added (with the appropriate availability), as long as it has a default implementation.
+
+This rule will be replaced by the following rule and be moved into the 'forbidden' section:
```

Comment:
Note that the document is listing *binary* compatible changes. The addition of an associated type requirement with a default implementation has always been and will remain a potentially *source-breaking* change.

Reply:
Thanks for pointing this out. That means that this proposal doesn't affect API resilience.

Also, I assume that ABI stability won't be broken as well.

Reply:
> Also, I assume that ABI stability won't be broken as well.

Yes, I don't think we'll even have to touch the ABI.

Reply:
I've changed ABI stability and API resilience to state that they're not going to be impacted by this proposal.